panther_battery/src/battery_publisher.cpp

```diff
@@ -0,0 +1,78 @@
+#include <panther_battery/battery_publisher.hpp>
+
+#include <memory>
+#include <stdexcept>
+
+#include <rclcpp/rclcpp.hpp>
+
+namespace panther_battery
+{
+
+BatteryPublisher::BatteryPublisher(std::shared_ptr<rclcpp::Node> node) : node_(node)
+{
+  node_->declare_parameter<float>("battery_timeout", 1.0);
+  battery_timeout_ = node_->get_parameter("battery_timeout").as_double();
+
+  charger_connected_ = false;
+  last_battery_info_time_ = rclcpp::Time(int64_t(0), RCL_ROS_TIME);
+
+  io_state_sub_ = node_->create_subscription<IOStateMsg>(
+    "hardware/io_state", 10,
```

Comment:
```suggestion
    "hardware/io_state", 3,
```
IOState is published so rarely, the queue size can be much smaller